src/doc/src/reference/specifying-dependencies.md

```diff
@@ -45,40 +59,40 @@ be allowed with them:
 0      :=  >=0.0.0, <1.0.0
 ```
 
-This compatibility convention is different from SemVer in the way it treats
-versions before 1.0.0. While SemVer says there is no compatibility before
-1.0.0, Cargo considers `0.x.y` to be compatible with `0.x.z`, where `y ≥ z`
-and `x > 0`.
+This matching strategy is different from SemVer in the way it treats
+versions before 1.0.0: While SemVer says there is no compatibility before
+1.0.0, Cargo considers `0.y.b` to be compatible with `0.y.a`, where `b ≥ a`
+and `y > 0`. The version `0.0.z` is not considered compatible with any
+other version.
 
 It is possible to further tweak the logic for selecting compatible versions
 using special operators, though it shouldn't be necessary most of the time.
 
-### Caret requirements
+### Tilde requirement
 
-**Caret requirements** are an alternative syntax for the default strategy,
-`^1.2.3` is exactly equivalent to `1.2.3`.
+**Tilde requirement** Example `~1.2.3`
 
-### Tilde requirements
-
-**Tilde requirements** specify a minimal version with some ability to update.
-If you specify a major, minor, and patch version or only a major and minor
-version, only patch-level changes are allowed. If you only specify a major
-version, then minor- and patch-level changes are allowed.
-
-`~1.2.3` is an example of a tilde requirement.
+For a tilde requirement, an update is allowed if the new version is newer and
+does not modify the right-most specified integer, unless it is a patch number.
+In other words, if you specify `x.y.z` then `x.y.b` is allowed if b>z. If you
+specify `x.y` then any `x.y.b` is allowed. If you specify `x` then any `x.a.b`
+is allowed.
 
 ```notrust
 ~1.2.3  := >=1.2.3, <1.3.0
 ~1.2    := >=1.2.0, <1.3.0
 ~1      := >=1.0.0, <2.0.0
 ```
 
-### Wildcard requirements
+### Wildcard requirement
 
-**Wildcard requirements** allow for any version where the wildcard is
-positioned.
+**Wildcard requirement** Examples: `*`, `1.*` and `1.2.*` are examples of wildcard
```

Comment:
I feel like something is lost by focusing on examples over the one line summary.  While people will need to know what it looks like to match the syntax they see in a `Cargo.toml` file, they also need a brief description of what it means

I think the ideal would be if we could be consistent on a one line description immediately followed by an example and then leave longer descriptions for after that,.

Reply:
I can flip the order. I tried to be consistent by provided an example immediately after each heading, which was then immediately followed by a short description of what the reader is looking at. Easy to flip the order.

Reply:
By the way, is there any situation where a wildcard requirement is in some sense superior to (e.g., can specify more than) a tilde requirement? They seem to be equivalent in that x.y.* is precisely ~x.y etc. The reverse is not true, since ~x.y.z means > z and there's not wildcard equivalent.

I'm asking because it may be helpful to state the equivalence. Alert readers will otherwise spend time wondering why both exist, and therefore, whether there's a use case they're not thinking of.